### PR TITLE
Cleanup for `peerAddresses` implementation

### PR DESCRIPTION
**Pull request by feross**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[ ] New feature
[x] Other, please explain: 

**What changes did you make? (Give an overview)**

Cleanup, remove duplication of the de-dupe code, change `x.pe` back to a string.

**Which issue (if any) does this pull request address?**

None

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -61,13 +61,6 @@ function magnetURIDecode (uri) {
     }
   })
 
-  if (result['x.pe']) {
-    // remove duplicates and convert to array
-    result['x.pe'] = Array.isArray(result['x.pe'])
-      ? Array.from(new Set(result['x.pe']))
-      : [result['x.pe']]
-  }
-
```
Comment (feross) [inline index.js:70]: The code for duplicate removal should only run for the friendly API -- `peerAddresses` -- just like we run it for `announce` but we leave `tr` as the raw value.

File: index.js [hunk 2]
```diff
@@ -85,11 +78,11 @@ function magnetURIDecode (uri) {
...
   if (result.kt) result.keywords = result.kt
-  if (result['x.pe']) result.peerAddresses = result['x.pe']
 
-  if (typeof result.tr === 'string') result.announce = [result.tr]
-  else if (Array.isArray(result.tr)) result.announce = result.tr
-  else result.announce = []
+  result.announce = []
+  if (typeof result.tr === 'string' || Array.isArray(result.tr)) {
+    result.announce = result.announce.concat(result.tr)
+  }
```
Comment (feross) [inline index.js:85]: This is just code cleanup. I made the style similar to the handling of `urlList` below.

File: index.js [hunk 3]
```diff
@@ -99,15 +92,15 @@ function magnetURIDecode (uri) {
...
+
   // remove duplicates by converting to Set and back
   result.announce = Array.from(new Set(result.announce))
   result.urlList = Array.from(new Set(result.urlList))
-
-  // convert x.pe into two level object
-  if (result['x.pe']) {
-    result.x = { pe: result['x.pe'] }
-    delete result['x.pe']
-  }
```
Comment (feross) [inline index.js:110]: I thought automatically turning x.pe into a nested object was clever but I don't think it's necessary. I renamed the nested x.pe object to the string 'x.pe' for a simpler implementation and less transformation back and forth.

The user is just going to use the friendly `peerAddresses` key anyway 👍

File: test/decode.js [hunk 4]
```diff
@@ -148,23 +148,26 @@ test('decode: select-only', t => {
...
 })
 
 test('decode: peer-address remove duplicates', t => {
   const result = magnet('magnet:?xt=urn:btih:64DZYZWMUAVLIWJUXGDIK4QGAAIN7SL6&x.pe=123.213.32.10:47450&x.pe=[2001:db8::2]:55013&x.pe=123.213.32.10:47450')
-  const peerAddresses = ['123.213.32.10:47450', '[2001:db8::2]:55013']
-  t.deepEqual(result.x.pe, peerAddresses)
-  t.deepEqual(result.peerAddresses, peerAddresses)
+
+  // raw value is *not* deduped
+  t.deepEqual(result['x.pe'], ['123.213.32.10:47450', '[2001:db8::2]:55013', '123.213.32.10:47450'])
```
Comment (feross) [inline test/decode.js:168]: Note how the raw `x.pe` value may be a string or an array, just like `tr`.

File: test/decode.js [hunk 5]
```diff
@@ -148,23 +148,26 @@ test('decode: select-only', t => {
...
+
+  // friendly value is deduped
+  t.deepEqual(result.peerAddresses, ['123.213.32.10:47450', '[2001:db8::2]:55013'])
```
Comment (feross) [inline test/decode.js:171]: But `peerAddresses` is always an array, just like `announce`.

**Review by hicom150 — APPROVED** [review]
Really nice cleanup and explanations! Thank you for sharing this ❤️

**Comment by feross** [comment]
Thank you for the review and merge, @hicom150!

**Comment by feross** [comment]
Released as 6.0.0 since we technically changed the API.

**Pull request merged**